File: 231_Power_of_Two.js
var isPowerOfTwo = function(n) {
    if(n === 1) return true;
    if(n % 2 !== 0) return false;
    let i = 2;
    while(i < n) i *= 2;
    if(i === n) return true;
    return false;
};

File: 231_Power_of_Two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n == 1){
            return true;
        }
        else if(n % 2 != 0){
            return false;
        }
        long int temp = 1;
        while(temp < n){
            temp *= 2;
        }
        if(temp == n){
            return true;
        } 
        else{
            return false;
        }
    }
};
